hdr_histogram.pc.in

```diff
@@ -6,7 +6,6 @@ Name: @PROJECT_NAME@
 Description: @CMAKE_PROJECT_DESCRIPTION@
 URL: @CMAKE_PROJECT_HOMEPAGE_URL@
 Version: @PROJECT_VERSION@
-Requires.private: @PC_REQUIRES_PRIVATE_ZLIB@
```

Comment:
It is more correct to use `Requires.private: zlib` than `Libs.private: -lz`.

@tristan957 why did you include both anyway?

And especially, why include a versioned requirement on zlib?

From #103

> I picked the minimum zlib version based on the Ubuntu 20.04 version of zlib. Happy to change it if you have a target in mind.

No, just... No. This is wrong. Why require the most recent version of a dependency "just as a matter of course"?

The `find_package(ZLIB)` does not specify a version and hence, per definition, the pkg-config file MUST not specify a more restrictive public export interface than its own link interface.

Reply:
@eli-schwartz see rewrite

Reply:
I confirm this fix allows to use generated pkgconfig file on old RHEL 7, so the version constraint was the main problem.

Reply:
Because, I am blind...

No where in the pkg-config spec did I see you could not specify a version

Reply:
Whether you use --exists or --cflags --libs it will fail if the package is invalid (for example its requirements cannot be met).

@tristan957, version requirements are optional (just like they are for meson `dependency('zlib', version: '>=1.2.11')` where you don't need to specify the version kwarg) but if specified they must be met.

Hence they should match the build system lookup, and be unversioned if the build system looks up an unversioned ZLIB, or versioned if the build system requires a specific version of ZLIB.

The nice thing about meson is its pkg-config generator handles all this automatically for you... if you add a pkgconfig dependency to a generated .pc file then it includes the version restriction only if the original lookup specified a meson.build `version: ...` kwarg.

Anyway it's nice either way to have a pkg-config file, they can always be fixed if they have minor bugs in them. :)

Reply:
Yes. Meson is my crutch. Love that this is all done for me. I tried to find something similar for CMake, but was unsuccessful as you can see :D.

Reply:
Also: I did after all suggest you add the pkg-config file to begin with, so you're more than welcome to `@` me for review on the thing I suggested. Even though this isn't using the Meson build system. ;)